spec/features/copy_reaction_spec.rb

```diff
@@ -2,115 +2,107 @@
+  let(:reaction1) { create(:reaction, status: 'Successful', short_label: 'Reaction1') }
 
   let(:material2) { create(:sample, name: 'Material2', target_amount_value: 7.15, molecule: m1) }
   let(:product2) { create(:sample, name: 'Product2', real_amount_value: 4.671, molecule: m2) }
   let(:reaction2) { create(:reaction, status: 'Successful', short_label: 'Reaction2') }
 
-  let(:material3) { create(:sample, name: 'Material3', target_amount_value: 7.15, molecule: m1) }
-  let(:product3) { create(:sample, name: 'Product3', real_amount_value: 4.671, molecule: m2) }
-  let(:reaction3) { create(:reaction, status: 'Successful', short_label: 'Reaction3') }
-
-  let!(:col) { create(:collection, user_id: user.id, label: 'Col1') }
-
-  let!(:root_share) { create(:collection, user: user, shared_by_id: user2.id, is_shared: true, is_locked: true) }
-  let!(:cshare) { create(:collection, user: user, label: 'share-col', permission_level: 10, sample_detail_level: 10, reaction_detail_level: 10, shared_by_id: user2.id, is_shared: true, ancestry: root_share.id.to_s) }
-  let!(:cshare2) { create(:collection, user: user, label: 'share-col-2', permission_level: 10, sample_detail_level: 0, reaction_detail_level: 0, shared_by_id: user2.id, is_shared: true, ancestry: root_share.id.to_s) }
-
-  before do
-    sign_in(user)
-    fp = Rails.public_path.join('images', 'molecules', 'molecule.svg')
-    svg_path = Rails.root.join('spec', 'fixtures', 'images', 'molecule.svg')
-    `ln -s #{svg_path} #{fp} ` unless File.exist?(fp)
-
-    CollectionsSample.find_or_create_by!(sample: material, collection: col)
-    CollectionsSample.find_or_create_by!(sample: product, collection: col)
-    CollectionsReaction.find_or_create_by!(reaction: reaction, collection: col)
-    ReactionsStartingMaterialSample.create!(reaction: reaction, sample: material, reference: true, equivalent: 1)
-    ReactionsProductSample.create!(reaction: reaction, sample: product, equivalent: 1)
-
-    CollectionsSample.find_or_create_by!(sample: material2, collection: cshare)
-    CollectionsSample.find_or_create_by!(sample: product2, collection: cshare)
-    CollectionsReaction.find_or_create_by!(reaction: reaction2, collection: cshare)
-    ReactionsStartingMaterialSample.create!(reaction: reaction2, sample: material2, reference: true, equivalent: 1)
-    ReactionsProductSample.create!(reaction: reaction2, sample: product2, equivalent: 1)
-
-    CollectionsSample.find_or_create_by!(sample: material3, collection: cshare2)
-    CollectionsSample.find_or_create_by!(sample: product3, collection: cshare2)
-    CollectionsReaction.find_or_create_by!(reaction: reaction3, collection: cshare2)
-    ReactionsStartingMaterialSample.create!(reaction: reaction3, sample: material3, reference: true, equivalent: 1)
-    ReactionsProductSample.create!(reaction: reaction3, sample: product3, equivalent: 1)
-
-  end
-
-  it ' new reaction', js: true do
```

Comment:
is this one gone?

Reply:
Yes, I removed it because as far as I can tell it only asserts that the reaction detail dialog isn't open. This assertion is a bit redundant because it's also (implicitely) tested in the other `it` blocks.

Reply:
I think the reactionDetail panel shoulb be  there, but because the reaction is new and not existing yet in the DB, there is no option to copy it

Reply:
right I see, something has been missing all along.
I think the `first('i.icon-reaction').click`  line should have been duplicated and run twice.
- 1st click is to switch the list pannel to display the reaction list => that will  change the add-element button on the page-top action bar to an add-reaction button  => that Button will then contain the first(i.icon-reaction ) of the page
- 2nd click is to build a new reaction => new built dont have the copy button until after created (the purpose of the this test I think)
I though the current line first('i.icon-reaction').clik was targetted at the reaction icon to add a reaction. But this one is not visible unless you are on the reaction list. so as of now this test is unlikely to ever fail  .